scripts/copy-kubeconfig.yaml

```diff
@@ -0,0 +1,61 @@
+---
+- hosts: kube-master[0]
+  gather_facts: no
+  become: yes
+  tasks:
+  - fetch:
+      src: "/etc/kubernetes/ssl/{{ item }}.pem"
+      dest: "{{ playbook_dir }}/kubectl/{{ item }}.pem"
+      flat: True
+    with_items:
+      - admin-{{ inventory_hostname }}-key
+      - admin-{{ inventory_hostname }}
+      - ca
+  - name: export hostname
+    set_fact:
+      kubectl_name: "{{ inventory_hostname }}"
+
+- hosts: localhost
+  connection: local
+  vars:
+    kubectl_name: "{{ hostvars[groups['kube-master'][0]].kubectl_name }}"
+    cluster_name: "{{ hostvars[groups['kube-master'][0]].cluster_name }}"
+    kube_apiserver_port: "{{ hostvars[groups['kube-master'][0]].kube_apiserver_port }}"
+    system_namespace: "{{ hostvars[groups['kube-master'][0]].system_namespace }}"
+  tasks:
+  - name: "check if context admin@{{ cluster_name }} exists"
+    command: kubectl config get-contexts admin@{{ cluster_name }}
+    register: kctl
+    failed_when: kctl.rc == 0
+
+  - block:
+    - name: "create cluster {{ cluster_name }}"
+      command: >
+        kubectl config set-cluster {{ cluster_name }}
+        --server=https://{{ kubectl_name }}:{{ kube_apiserver_port }}
```

Comment:
I concur. If the hostname is not resolvable, you will not be able to reach it with Ansible.